core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -177,4 +177,11 @@ trait EntityDecoderInstances {
 
   implicit def multipart: EntityDecoder[Multipart] =
     MultipartDecoder.decoder
+
+  /** An entity decoder that ignores the content and returns unit. */
+  implicit val void: EntityDecoder[Unit] = new EntityDecoder {
+    val consumes: Set[MediaRange] = Set.empty
+
+    def decode(msg: Message, strict: Boolean): DecodeResult[Unit] = DecodeResult.success(())
```

Comment:
Upon thinking about it, I think killing the body seems the right thing to do: the server/client can decide how to handle kill gracefully and middleware will still be run in the unfortunate case that it has meaningful side effects.